pkg/scheduler/core/generic_scheduler.go

```diff
@@ -953,6 +953,7 @@ func (g *genericScheduler) selectVictimsOnNode(
 		}
 		status := prof.RunPreFilterExtensionRemovePod(ctx, state, pod, rp, nodeInfo)
 		if !status.IsSuccess() {
+			nodeInfo.AddPod(rp)
```

Comment:
Feel like this will loop back to this pod in the loop at line 973. Since the AddPod() appends the pod to the end of nodeInfo.Pods again. But I maybe wrong.